runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/environment/DockerEnvironmentFactory.java

```diff
@@ -152,8 +152,10 @@ public RemoteEnvironment createEnvironment(Environment environment) throws Excep
       LOG.debug("Created Docker Container with Container ID {}", containerId);
       // Wait on a client from the gRPC server.
       while (instructionHandler == null) {
+        Preconditions.checkArgument(
+            docker.isContainerRunning(containerId), "No container running for id " + containerId);
         try {
-          instructionHandler = clientSource.take(workerId, Duration.ofMinutes(2));
+          instructionHandler = clientSource.take(workerId, Duration.ofSeconds(30));
```

Comment:
Are there perhaps environments that need a longer time to spin up the services? Would only lower to 1 minute.

Reply:
1 Min sounds good to me.